src/btreemap/node.rs

```diff
@@ -413,3 +424,61 @@ enum Value {
     // The value's offset in the node.
     ByRef(Bytes),
 }
+
+/// Stores version-specific data.
+#[derive(Debug, PartialEq, Copy, Clone, Eq)]
+pub enum Version {
+    /// V1 nodes must have bounds for the keys and values.
+    V1 {
+        max_key_size: u32,
+        max_value_size: u32,
+    },
+    /// V2 nodes have a fixed page size.
+    V2(PageSize),
+}
+
+impl Version {
+    fn page_size(&self) -> u32 {
```

Comment:
Gotcha. In that case maybe we need to find a way to encode that V1 also has a `PageSize` concept. I think the disconnect for me was that in the `Version` type, `V1` does not have an associated `PageSize` like `V2` does. Maybe we actually use a `PageSize` there as well (I suppose it would be the variant `PageSize::Derived` in that case). Not entirely sure this makes sense or is indeed better but thought I'd throw it out there.

Reply:
I see your point, and it does makes sense. I introduced a `DerivedPageSize` type to make this more clear per your suggestion.